torch/_inductor/compile_fx.py

```diff
@@ -404,25 +405,37 @@ def static_input(x):
     return torch.as_strided(buffer, x.size(), x.stride())
 
 
+def index_expanded_dims_and_copy_(dst, src, expanded_dims):
+    "Index into expanded dimensions of both dst and src then copy_"
+    dst = index_expanded_dims(dst, expanded_dims)
+    src = index_expanded_dims(src, expanded_dims)
+    dst.copy_(src)
+
+
 def cudagraphify_impl(model, inputs, static_input_idxs=()):
     """
     Assumes inputs[static_input_idxs[i]] are always the same memory address
     """
     static_input_idxs = remove_unaligned_input_idxs(inputs, static_input_idxs)
 
     assert isinstance(inputs, (list, tuple))
-    static_inputs = [
-        static_input(x).copy_(x.detach())
-        if idx not in static_input_idxs
-        else x.detach()
-        for idx, x in enumerate(inputs)
-    ]
 
     inps_expanded_dims = [
         get_expanded_dims(x) if idx not in static_input_idxs else []
         for idx, x in enumerate(inputs)
     ]
 
+    # allocate static tensor inputs
+    static_inputs = [
+        static_input(x) if idx not in static_input_idxs else x.detach()
+        for idx, x in enumerate(inputs)
+    ]
+
+    # copy over input values for fresh allocations
+    for idx, (x, expanded_dims) in enumerate(zip(inputs, expanded_dims)):
```

Comment:
I got `UnboundLocalError: local variable 'expanded_dims' referenced before assignment` on this line, maybe we want:

for idx, (x, expanded_dims) in enumerate(zip(inputs, **inps_expanded_dims**)):

Reply:
and a test that would catch it